### ISSUE TITLE
Can't find where you add regularization term to your total loss

### ISSUE THREAD

**Issue by dvornikita** [endernewton/tf-faster-rcnn#217]
I've inspected the code and from what I understand, you define the regularizers but don't add the regularization term to your total_loss that is being optimized, which is just the sum of 4 losses defined in add_losses (I checked this). Do I understand correctly that there is no regularization during the training?

**Comment by endernewton** [comment]
slim provides regularization. please check the network.py file.

**Comment by dvornikita** [comment]
in the network.py you do define the regularization in the arg_scope but the total loss is defined as
      loss = cross_entropy + loss_box + rpn_cross_entropy + rpn_loss_box
which doesn't include regularization. 
And if I understand correctly, to optimize it with regularization you need to do something like
loss += tf.get_collection(tf.GraphKeys.REGULARIZATION_LOSSES)
and minimizing the loss without this addition would mean no regularization.

**Comment by endernewton** [comment]
ahh i see, my mistake! thanks so much for catching this bug! do you want to submit a pull request? I can also test how much it affects the performance then.

**Comment by endernewton** [comment]
fixed myself. closing

**Comment by yaoqi-zd** [comment]
@endernewton I'm a little curious that the performance seems have little difference with and without the regularization loss, does that make sense? regularization seems essential for neural network training in my mind.

**Comment by yaoqi-zd** [comment]
I'm curious why the network without regularization still performs so well, Did you test how much regularization affects the performance?

**Comment by gwspotex** [comment]
This paper show how "[Dropout Training as Adaptive Regularization](https://papers.nips.cc/paper/4882-dropout-training-as-adaptive-regularization.pdf)",maybe can give you answer
